build/Dockerfile

```diff
@@ -1,7 +1,66 @@
-FROM golang:1.7.3
-RUN apt-get update && apt-get install -y libpcap-dev python-requests time file
-RUN go get github.com/golang/lint/golint github.com/fzipp/gocyclo github.com/client9/misspell/cmd/misspell
-RUN go clean -i net && go install -tags netgo std
-RUN go install -race -tags netgo std
+FROM golang:1.7.4
+
+ENV GOARM 7
+ENV DEB_CROSSPLATFORMS \
+	armhf \
+	arm64
+
+ENV GO_CROSSPLATFORMS \
+	linux/arm \
+	linux/arm64
+
+ENV GCC_CROSSCOMPILERS \
+	arm-linux-gnueabi \
+	aarch64-linux-gnu
+
+RUN for platform in ${GO_CROSSPLATFORMS}; do GOOS=${platform%/*} GOARCH=${platform##*/} go install -tags netgo std; done
+
+RUN echo "deb http://archive.ubuntu.com/ubuntu xenial main universe" > /etc/apt/sources.list.d/cgocrosscompiling.list \
+  && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 40976EAF437D05B5 3B4FE6ACC0B21F32 \
+  && apt-get update \
+  && apt-get install -y build-essential \
+  && for platform in ${DEB_CROSSPLATFORMS}; do apt-get install -y crossbuild-essential-${platform}; done \
+  && rm /etc/apt/sources.list.d/cgocrosscompiling.list \
+  && apt-get update \
+  && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+RUN apt-get update \
+	&& apt-get install -y \
+		libpcap-dev \
+		python-requests \
+		time \
+		file \
+		flex \
+		bison
+
+RUN go get \
+	github.com/golang/lint/golint \
+	github.com/fzipp/gocyclo \
+	github.com/client9/misspell/cmd/misspell
+
+RUN go clean -i net \
+	&& go install -tags netgo std \
```

Comment:
Is it really necessary to run this twice?
First without race and then with race...